src/deploy/functions/runtimes/node/parseTriggers.ts

```diff
@@ -211,19 +211,40 @@ export function addResourcesToBackend(
       triggered = {
         eventTrigger: {
           eventType: annotation.eventTrigger!.eventType,
-          eventFilters: {
-            resource: annotation.eventTrigger!.resource,
-          },
+          eventFilters: [
+            {
+              attribute: "resource",
+              value: annotation.eventTrigger!.resource,
+            },
+          ],
           retry: !!annotation.failurePolicy,
         },
       };
 
       // TODO: yank this edge case for a v2 trigger on the pre-container contract
       // once we use container contract for the functionsv2 experiment.
-      if (STORAGE_V2_EVENTS.find((event) => event === (annotation.eventTrigger?.eventType || ""))) {
-        triggered.eventTrigger.eventFilters = {
-          bucket: annotation.eventTrigger!.resource,
-        };
+      if (annotation.platform === "gcfv2") {
+        if (annotation.eventTrigger!.eventType === v2events.PUBSUB_PUBLISH_EVENT) {
+          triggered.eventTrigger.eventFilters = [
+            {
+              attribute: "topic",
+              value: annotation.eventTrigger!.resource,
+            },
+          ];
+        }
+
+        if (
+          v2events.STORAGE_EVENTS.find(
+            (event) => event === (annotation.eventTrigger?.eventType || "")
```

Comment:
nit: I don't think you need the `|| ""` because you shouldn't crash comparing a string to `undefined`